### PR TITLE
Update tranche-space dashboard

### PR DESCRIPTION
**Pull request by Siddharth2207**
## Motivation
Binding and charting buys and sells when twap identity is used for TKN tranche ratio denominations.

## Solution
Add bindings for `tranche-space` when tranche ratio/base are denominated in token like WETH.  Update the `plottables` binding, so the effective price denominated in USD can be plotted in the chart for when the twap identity is used for buys and sells.

## Checks

- [x] Update bindings to have twap identity for buy and sell
- [x] Update `plottables` binding to have effective price denominated in USD on the stack.

### CODE REVIEW

File: src/tranche-space.rain [hunk 1]
```diff
@@ -390,12 +420,15 @@ deployments:
 #orderbook-subparser !The subparser for the Orderbook
 
 #plottables-test
+  tranche-io-ratio
   amount
   io-ratio:,
   input-amount: decimal18-mul(amount io-ratio),
-  effective-price: decimal18-inv(io-ratio);
+  effective-price: decimal18-inv(io-ratio),
+  effective-price-twap: decimal18-inv(tranche-io-ratio);
```
Comment (thedavidmeister) [inline src/tranche-space.rain:428]: this isn't always a twap

File: src/tranche-space.rain [hunk 2]
```diff
@@ -481,7 +514,7 @@ deployments:
   amount-available: decimal18-mul(tranche-total-size tranche-space-available),
   amount: if(amount-is-output amount-available decimal18-div(amount-available final-io-ratio)),
   io-ratio: final-io-ratio,
-  :call<'plottables>(amount io-ratio);
+  :call<'plottables>(tranche-io-ratio amount io-ratio);
```
Comment (thedavidmeister) [inline src/tranche-space.rain:517]: we should show the multiplier

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: src/tranche-space.rain (hunk 1 above)
Comment (Siddharth2207) [reply src/tranche-space.rain:428]: should i just rename a `effective-price-twap`

File: src/tranche-space.rain (hunk 2 above)
Comment (Siddharth2207) [reply src/tranche-space.rain:517]: sorry more context please, `tranche-io-ratio` is the multiplier what is it that needs to be shown ?

File: src/tranche-space.rain (hunk 1 above)
Comment (Siddharth2207) [reply src/tranche-space.rain:428]: pushed a fix .

File: src/tranche-space.rain (hunk 2 above)
Comment (Siddharth2207) [reply src/tranche-space.rain:517]: Added the multiplier metric.

File: src/tranche-space.rain [hunk 3]
```diff
@@ -129,19 +129,29 @@ scenarios:
...
+                # for e.g. a TWAP USDT based multiplier.
+                # Typically this is NOT needed as the tranche size and ratio ARE
+                # denominated in the input/output tokens.
+                io-ratio-multiplier: '''io-ratio-multiplier-identity'
+
+                # If TKN was being traded for some other token, for e.g WETH and if
+                # the tranche size/ratio was denominated in USD, then the reference and
+                # reserve bindings need to be given along with the twap bindings
+                reference-stable: 0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9
+                reference-stable-decimals: 6
```
Comment (thedavidmeister) [inline src/tranche-space.rain:149]: are the decimals needed?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: src/tranche-space.rain (hunk 3 above)
Comment (Siddharth2207) [reply src/tranche-space.rain]: yes, they are passed as an argument for the uniswap twap call.

**Comment by Siddharth2207** [comment]
Closing this PR and putting a new one.

**Pull request closed without merging**